api/v0/ingest/models/discover_request.go

```diff
@@ -0,0 +1,80 @@
+package models
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/filecoin-project/storetheindex/config"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p-core/record"
+)
+
+type DiscoverRequest struct {
+	ProviderID    peer.ID
+	DiscoveryAddr string
+	Seq           uint64
+}
+
+// DiscoverRequestEnvelopeDomain is the domain string used for discover requests contained in a Envelope
+const DiscoverRequestEnvelopeDomain = "indexer-discover-request-record"
```

Comment:
nit-pick: I don't know what is the standard in the libp2p world (I just found [this example from tests](https://github.com/libp2p/go-libp2p-core/blob/ea6cbf9ff2894c96aa014006c475207e01d6780a/record/record_test.go#L5)), but I think we should use a path format for domains: `indexer/discover/request/record`

Reply:
I was following the example here: https://github.com/libp2p/go-libp2p-core/blob/v0.9.0/peer/record.go#L23

:shrug: